### ISSUE TITLE
sendAndConfirm transaction: update how blockhash is set

### ISSUE THREAD

**Issue by mcintyre94** [coral-xyz/anchor#2091]
Currently in `sendAndConfirm` we use the deprecated `getRecentBlockhash` function:

https://github.dev/coral-xyz/anchor/blob/205e9d875f9abf1fc379cfc8114c5c53f0207141/ts/src/provider.ts#L136-L138
```
    tx.recentBlockhash = (
      await this.connection.getRecentBlockhash(opts.preflightCommitment)
    ).blockhash;
```

This can return the same blockhash for multiple transactions. This means that if you send the same transaction multiple times using Anchor (eg. with `program.methods.[...].rpc()`) you can get an error like "This transaction has already been processed"

Experimenting locally, if we moved to the non-deprecated replacement `getLatestBlockhash`, this doesn't happen - you get a different blockhash for each transaction. 

In addition, it's not possible to work around this by using `.transaction()` instead of `rpc()` and setting the blockhash yourself, because this code will overwrite it back to the older one returned by `getRecentBlockhash`

Proposed changes:

- Use `getLatestBlockhash`, which is the replacement recommended for the deprecated `getRecentBlockhash` and gives each transaction a unique blockhash. We could also set `tx.lastValidBlockHeight`, which is also returned by `getLatestBlockhash`

- Should we avoid setting `recentBlockhash` (and `lastValidBlockHeight`) if they've already been set on the input transaction? In the `.rpc()` case they haven't been. But this would provide a bit more flexibility if this logic changes again in future - we'd be able to manually update them as needed and then pass the transaction to `sendAndConfirm` without them being overwritten by the hardcoded logic.

More context: https://solana.stackexchange.com/questions/1161/how-to-avoid-sendtransactionerror-this-transaction-has-already-been-processed

I'd be happy to contribute a PR for this issue :)

**Comment by jsbarrett** [comment]
I ran into this tonight as well

Would like to see this get fixed ... in the meantime I'm having to just wait half a second before executing next call

**Comment by mcintyre94** [comment]
Closing - fixed by #2205
